File: validParentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    if(s === null || s.length === 0){
    	return true;
    }
    var check = [];
    for(var i = 0; i < s.length; i++){
    	if(s[i] === "(" || s[i] === "[" || s[i] === "{"){
    		check.push(s[i]);
    	}else if(s[i] === ")" && check.length !== 0 && check[check.length - 1] === "(" ){
    		check.pop();
    	}else if(s[i] === "]" && check.length !== 0 && check[check.length - 1] === "[" ){
    		check.pop();
    	}else if(s[i] === "}" && check.length !== 0 && check[check.length - 1] === "{" ){
    		check.pop();
    	}else{
    		return false;
    	}
    }

    if(check.length === 0){
    	return true;
    }else{
    	return false;
    }
};

console.log(isValid("{(())}[()]"));

File: valid_parentheses.rb
# @param {String} s
# @return {Boolean}
def is_valid(s)
  if s.nil? || s.length === 0
    return true
  end 

  stack = []

  for i in 0...s.length
    if s[i] === "(" || s[i] === "{" || s[i] === "["
      stack.push(s[i])
    elsif stack.length != 0 && stack[-1] === "(" && s[i] === ")"
      stack.pop
    elsif stack.length != 0 && stack[-1] === "[" && s[i] === "]"
      stack.pop
    elsif stack.length != 0 && stack[-1] === "{" && s[i] === "}"
      stack.pop
    else
      return false
    end
  end

  if stack.length === 0
    return true
  else
    return false
  end
  
end